imagetest/test_suites/storage_perf/iops_print_test.go

```diff
@@ -4,10 +4,131 @@
+	X map[string]interface{} `json:"-"`
+}
+
+func getMountDiskPartition(diskExpectedSizeGb int) (string, error) {
+	lsblkCmd := "lsblk"
+	if !utils.CheckLinuxCmdExists(lsblkCmd) {
+		return "", fmt.Errorf("could not find lsblk")
+	}
+	lsblkout, err := exec.Command(lsblkCmd, "-o", "name,size,type", "--json").Output()
+	if err != nil {
+		return "", fmt.Errorf("failed to execute lsblk: %v", err)
+	}
+
+	var blockDevices BlockDeviceList
+	if err := json.Unmarshal(lsblkout, &blockDevices); err != nil {
+		return "", fmt.Errorf("failed to unmarshal lsblk output: %v", err)
+	}
+
+	diskExpectedSizeGbString := strconv.Itoa(diskExpectedSizeGb) + "G"
+	for _, blockDev := range blockDevices.BlockDevices {
+		if strings.ToLower(blockDev.Type) == "disk" && blockDev.Size == diskExpectedSizeGbString {
+			return blockDev.Name, nil
+		}
+	}
+
+	return "", fmt.Errorf("disk block with size not found")
+}
+
+func installFio() error {
+	var installFioCmd *exec.Cmd
+	if utils.CheckLinuxCmdExists("apt") {
+		installFioCmd = exec.Command("apt", "install", "-y", "fio")
+	} else if utils.CheckLinuxCmdExists("yum") {
+		installFioCmd = exec.Command("yum", "-y", "install", "fio")
+	} else if utils.CheckLinuxCmdExists("zypper") {
+		installFioCmd = exec.Command("zypper", "--non-interactive", "install", "fio")
+	} else if utils.CheckLinuxCmdExists("dnf") {
+		installFioCmd = exec.Command("dnf", "-y", "install", "fio")
+	} else {
+		return fmt.Errorf("could not find package manager to install fio")
+	}
+
+	if err := installFioCmd.Run(); err != nil {
+		return fmt.Errorf("install fio cmd %s failed with error: %v", installFioCmd.Path, err)
+	}
+
+	return nil
+}
+
+func getMountDiskPartitionSymlink() (string, error) {
+	mountDiskSymlink := "/dev/disk/by-id/google-" + mountDiskName
+	symlinkRealPath, err := filepath.EvalSymlinks(mountDiskSymlink)
+	if err != nil {
+		return "", fmt.Errorf("symlink could not be resolved: %v", err)
+	}
+	return symlinkRealPath, nil
+}
+
 // TestIOPSPrint is a placeholder test which prints out info about iops
 func TestIOPSPrint(t *testing.T) {
+	//mountDiskSymlink := "/dev/disk/by-id/google-" + mountDiskName
+	//symlinkRealPath, err := filepath.EvalSymlinks(mountDiskSymlink)
+	//if err != nil {
+	//	t.Fatalf("symlink could not be resolved: %v", err)
+	//}
+
+	// try multiple methods to find the disk partition: if any succeeds, then continue.
+	symlinkRealPath := ""
+	diskPartition, err := getMountDiskPartition(HyperdiskSize)
+	if err == nil {
+		symlinkRealPath = "/dev/" + diskPartition
+	} else {
+		t.Logf("error with lsblk: %v", err)
+		symlinkRealPath, err = getMountDiskPartitionSymlink()
+		if err != nil {
+			t.Fatalf("failed to find symlink to mount disk with any method: error %v", err)
+		}
+	}
+
+	if !utils.CheckLinuxCmdExists(mkfsCmd) {
+		t.Fatalf("could not format mount disk: %s cmd not found", mkfsCmd)
+	}
+	mkfsFullCmd := exec.Command(mkfsCmd, "-m", "0", "-E", "lazy_itable_init=0,lazy_journal_init=0,discard", symlinkRealPath)
```

Comment:
There are two ways to benchmark performance- one is against the raw block device, other is against a mounted filesystem.  We should benchmark against the block device (so that the filesystem is not part of the equation). Filesystem types, configurations, options, and even versions can differ greatly. For our purpose we should look at the block device performance: https://cloud.google.com/compute/docs/disks/benchmarking-pd-performance#raw-disk

Reply:
changed to use the filesystem command